grails-app/controllers/au/org/emii/portal/DownloadController.groovy

```diff
@@ -174,10 +174,14 @@ class DownloadController extends RequestProxyingController {
                 if (fieldIndex < currentRow.length) {
 
                     def rowValue = currentRow[fieldIndex].trim()
-                    rowValue = rowValue.replace(prefixToRemove, newUrlBase)
+                    urlSubstitutions.each {
+                        search, replace ->
+
+                        rowValue = rowValue.replaceAll(search, replace)
+                    }
 
                     if (rowValue && includedUrls.add(rowValue)) {
-                        outputWriter.print "$rowValue\n"
```

Comment:
I think this used `print()` with a newline character for a reason. Maybe different platforms were using `\n\r` and that caused problems somewhere along the line (e.g. maybe GoGoDuck).

Happy to say that if we didn't have a test for it then it can't have been *that* important. But just something to keep in mind.